node/src/components/contract_runtime.rs

```diff
@@ -412,16 +439,19 @@ where
                 );
                 let engine_state = Arc::clone(&self.engine_state);
                 let metrics = Arc::clone(&self.metrics);
-                tokio::task::unconstrained(async move {
-                    let result = execute_finalized_block(
-                        engine_state.as_ref(),
-                        Some(metrics),
-                        protocol_version,
-                        execution_pre_state,
-                        finalized_block,
-                        deploys,
-                        transfers,
-                    );
+                tokio::task::spawn(async move {
```

Comment:
I believe we can get rid of this `spawn`, the `.ignore` here is currently called on the `JoinHandle`, as far as I can tell, so just call it on the `async` block directly?

```rust
                async move {
                    let result = run_intensive_task(move || {
                        execute_finalized_block(
                            engine_state.as_ref(),
                            Some(metrics),
                            protocol_version,
                            execution_pre_state,
                            finalized_block,
                            deploys,
                            transfers,
                        )
                    })
                    .await;
                    trace!(?result, "execute block response");
                    responder.respond(result).await
                }
                .ignore()
```

should just work?

Reply:
You are right. It's fixed.